solr/core/src/test/org/apache/solr/hamcrest/QueryMatchers.java

```diff
@@ -0,0 +1,184 @@
+package org.apache.solr.hamcrest;
+
+import org.apache.lucene.index.Term;
+import org.apache.lucene.search.BooleanClause;
+import org.apache.lucene.search.BooleanQuery;
+import org.apache.lucene.search.BoostQuery;
+import org.apache.lucene.search.DisjunctionMaxQuery;
+import org.apache.lucene.search.PhraseQuery;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.TermQuery;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.TypeSafeDiagnosingMatcher;
+
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.is;
+
+public class QueryMatchers {
+  private QueryMatchers() {}
+
+  public static Matcher<Query> termQuery(String field, String text) {
```

Comment:
Matcher factories are generally written as predicates, so `isTermQuery` rather than just `termQuery`.  But I suppose that makes nested definitions read a bit weird, so maybe this is fine.

Reply:
Hmm... let me think on how these should read.

assertThat q is disjunctionOf (termQuery, termQuery, termQuery)
assertThat q is boosted phraseQuery
assertThat q is booleanQueryOf( shouldClause termQuery, mustClause termQuery)

I like how the first two read but not quite satisfied with the last one. The `is` should be left out of the method name because callers can always add it where it grammatically makes sense.